### PR TITLE
Document s3 compatible storage docker compose 

### LINKED ISSUE #939: Document configuring the docker compose cluster to use external s3 services instead of minio
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/10074

### Which part(s) of the docs might be affected or should be updated? And how?

https://www.risingwave.dev/docs/current/risingwave-docker-compose/#minio

The minio section can be marked as optional. Instead, we can add another section that describes how to connect to an external s3.

### Reference

_No response_

### PR DESCRIPTION
**Pull request by wcy-fdu**
close https://github.com/risingwavelabs/risingwave-docs/issues/939
## Info
- **Description**: 
[ What's changed? ]

- **Preview**: 
[ Paste the preview link to the edited page here. Edit this item after you submit the PR and the preview is ready. ]

- **Related code PR**: 
[ Provide the link to the code PR here. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves [ Provide the link to the doc issue here. ]

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/deploy/risingwave-docker-compose.md [hunk 1]
```diff
@@ -40,11 +40,21 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
+### Start a cluster on minio
+The default storage backend is minio, to start the cluster, 
+
 ```shell
 cd docker
 docker compose up -d
 ```
+### Start a cluster on other s3 compatible object storage
+
+If you want to deploy RisingWave on other s3 compatible storage, fill in your `bucket name` in `risingwave/docker/docker-compose-s3-compatible.yml`, and config the `region`, `endpoint`, `access key` and `secret key` in risingwave/docker/s3-compatible.env (note that the `endpoint` cannot contain bucket-name). Then
```
Comment (wcy-fdu) [inline docs/deploy/risingwave-docker-compose.md]: Will add code link after https://github.com/risingwavelabs/risingwave/pull/10074 merged.

**Comment by hengm3467** [comment]
@neverchanje Please review this PR. Thanks.

**Comment by wcy-fdu** [comment]
> @neverchanje Please review this PR. Thanks.

~~Recently there are some minor changes of this part, I will update this PR later.~~

File: docs/deploy/risingwave-docker-compose.md [hunk 2]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
+### Start a cluster on MinIO
+
+To use MinIO as the storage backend, run:
+
 ```shell
 cd docker
 docker compose up -d
 ```
 
+### Start a cluster on other s3 compatible object storage
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
### Start a cluster on other S3-compatible object storage systems
```

File: docs/deploy/risingwave-docker-compose.md [hunk 3]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
 
 Now run the following commands to navigate to the `docker` directory and start the cluster from the pre-defined docker-compose file.
 
+### Start a cluster on MinIO
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
### Start a RisingWave cluster using MinIO
```

File: docs/deploy/risingwave-docker-compose.md [hunk 4]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
 After RisingWave is up and running, you need to connect to it via the Postgres interactive terminal `psql` so that you can issue queries to RisingWave and see the query results.
 
-
 ```shell
 psql -h localhost -p 4566 -d dev -U root
 ```
-    
-You can now [connect a streaming source to RisingWave](sql/commands/sql-create-source.md) and [issue SQL queries to manage your data](risingwave-sql-101.md).
 
+You can now [connect a streaming source to RisingWave](sql/commands/sql-create-source.md) and [issue SQL queries to manage your data](risingwave-sql-101.md).
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
You can now [connect a streaming source to RisingWave](/sql/commands/sql-create-source.md) and [issue SQL queries to manage your data](/risingwave-sql-101.md).
```

File: docs/deploy/risingwave-docker-compose.md [hunk 5]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
+To use MinIO as the storage backend, run:
+
 ```shell
 cd docker
 docker compose up -d
 ```
 
+### Start a cluster on other s3 compatible object storage
+
+If you want to deploy RisingWave on other s3 compatible storage, you need to customize these configurations.
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
If you want to deploy RisingWave using other S3-compatible storage systems, you need to customize these configurations.
```

File: docs/deploy/risingwave-docker-compose.md [hunk 6]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
+
+* Fill in your `bucket-name` [here](https://github.com/risingwavelabs/risingwave/blob/09cfe81dc8e230969b7b2e6f4883feb59f59a186/docker/docker-compose-s3.yml#L194).
+* Config the `region`, `endpoint`, `access key` and `secret key` in [risingwave/docker/aws.env](https://github.com/risingwavelabs/risingwave/blob/main/docker/aws.env). Note that the `endpoint` cannot contain bucket names.
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
* Configure `region`, `endpoint`, `access key`, and `secret key` in [risingwave/docker/aws.env](https://github.com/risingwavelabs/risingwave/blob/main/docker/aws.env). Note that `endpoint` cannot contain bucket names.
```

File: docs/deploy/risingwave-docker-compose.md [hunk 7]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
 After RisingWave is up and running, you need to connect to it via the Postgres interactive terminal `psql` so that you can issue queries to RisingWave and see the query results.
 
-
 ```shell
 psql -h localhost -p 4566 -d dev -U root
 ```
-    
-You can now [connect a streaming source to RisingWave](sql/commands/sql-create-source.md) and [issue SQL queries to manage your data](risingwave-sql-101.md).
 
+You can now [connect a streaming source to RisingWave](/sql/commands/sql-create-source.md) and [issue SQL queries to manage your data](/risingwave-sql-101.md).
```
Comment (CharlieSYH) [inline docs/deploy/risingwave-docker-compose.md]:
```suggestion
Now you can ingest and transform streaming data. See [Quick start](/get-started.md) for details.
```

File: docs/deploy/risingwave-docker-compose.md [hunk 8]
```diff
@@ -40,23 +40,38 @@ git clone https://github.com/risingwavelabs/risingwave.git
...
 ```shell
 cd docker
 docker compose up -d
 ```
 
+### Start a cluster on other S3-compatible object storage systems
+
+If you want to deploy RisingWave using other S3-compatible storage systems, you need to customize these configurations.
+
+* Fill in your `bucket-name` [here](https://github.com/risingwavelabs/risingwave/blob/09cfe81dc8e230969b7b2e6f4883feb59f59a186/docker/docker-compose-s3.yml#L194).
```
Comment (hengm3467) [inline docs/deploy/risingwave-docker-compose.md]: @CharlieSYH I feel [here] might not be the best way to present the info. I may make people feel that they need to go into the public github folder, but it actually means is to edit the docker-compose-s3.yml file that in the downloaded directory, and the link may be obsolete soon (or already). @wcy-fdu Am I correct?

Comment (wcy-fdu) [reply docs/deploy/risingwave-docker-compose.md]: I think the link is stable now, and in the future all s3&s3-compatible storage will use this file. Anyways, feel free to edit or fix, personally I think this pr can be merged.

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**